### PR TITLE
Update docs to point to new sdpa_kernel context manager

### PR DESCRIPTION
**Pull request by drisspg**
# Summary

Updates the SDPA docs to fix some small inaccuracies and points to the new sdpa_kernel context manger. The Enum like type binded from cpp SDPBackend does not render its fields for some reason. Manually list them instead for now

cc @svekars @brycebortree

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -5009,14 +5009,14 @@ def scaled_dot_product_attention(query, key, value, attn_mask=None, dropout_p=0.
...
     The context manager is the preferred mechanism:
 
-        - :func:`torch.backends.cuda.sdp_kernel`: A context manager used to enable/disable any of the implementations.
-        - :func:`torch.backends.cuda.enable_flash_sdp`: Enables or Disables FlashAttention.
-        - :func:`torch.backends.cuda.enable_mem_efficient_sdp`: Enables or Disables Memory-Efficient Attention.
-        - :func:`torch.backends.cuda.enable_math_sdp`: Enables or Disables the PyTorch C++ implementation.
+        - :func:`torch.nn.attention.sdpa_kernel`: A context manager used to enable/disable any of the implementations.
+        - :func:`torch.backends.cuda.enable_flash_sdp`: Globally Enables or Disables FlashAttention.
+        - :func:`torch.backends.cuda.enable_mem_efficient_sdp`: Globally Enables or Disables Memory-Efficient Attention.
+        - :func:`torch.backends.cuda.enable_math_sdp`: Globally Enables or Disables the PyTorch C++ implementation.
```
Comment (mikaylagawarecki) [inline torch/nn/functional.py:5015]: tiny nit:
```suggestion
        - :func:`torch.backends.cuda.enable_flash_sdp`: Globally enables or disables FlashAttention.
        - :func:`torch.backends.cuda.enable_mem_efficient_sdp`: Globally enables or disables Memory-Efficient Attention.
        - :func:`torch.backends.cuda.enable_math_sdp`: Globally enables or disables the PyTorch C++ implementation.
```

File: torch/nn/attention/__init__.py [hunk 2]
```diff
@@ -34,7 +34,15 @@
...
+    The following Enums are available:
+        - ERROR: An error occurred when trying to determine the backend.
+        - MATH: The math backend for scaled dot product attention.
+        - FLASH_ATTENTION: The flash attention backend for scaled dot product attention.
+        - EFFICIENT_ATTENTION: The efficient attention backend for scaled dot product attention.
+        - CUDNN_ATTENTION: The cuDNN backend for scaled dot product attention.
+
+    See :func:`torch.nn.attention.sdpa_kernel` for more details.
 
     ... warning:: This class is in beta and subject to change.
```
Comment (mikaylagawarecki) [inline torch/nn/attention/__init__.py:47]: nit: Rendering seems broken for the warning :/

```suggestion
    .. warning:: This class is in beta and subject to change.
```

**Review by mikaylagawarecki — APPROVED** [review]

File: torch/nn/attention/__init__.py (hunk 2 above)
Comment (drisspg) [reply torch/nn/attention/__init__.py:47]: Awesome, thanks for taking a look I will update

**Pull request closed without merging**